Bottom: 93c7ced34ad16d864bcdeaa48a06be24c76f9e45
Top:    abb11c53e0f101d8b1119d5374443185fa72c1b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:24:23 +0000

Turn on some debugging for intersections..

Have seen some use-after free corruption in a few places, and this was
aimed at testing whether or not the culprit was performing multiple
intersections on the same segment. (When we perform the deferred work
to split that segment, any later intersections will not point at the
correct seg structure - which has been free'd during the first splitting.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index adb33e4..3ef2e74 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,6 +39,7 @@
       are marked
 */
 
+#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -971,6 +972,7 @@ insert_vertex_in_seg (struct info *i, struct seg *s, Vector v, double param)
   DEBUGP ("new intersection on segment \"i\" at %#mD\n", v[0], v[1]);
 #endif
   i->node_insert_list = prepend_insert_node_task (i->node_insert_list, s, new_node, param);
+  assert (!s->intersected);
   s->intersected = 1;
   return true;
 }
